### ISSUE TITLE
Drying rack meat crash

### ISSUE THREAD

**Issue by jkramr** [penguin0616/insight#8]
Got the trace with the drying rack crash. Before it would crash when placing food on the rack. This time I just loaded while it just  WAS THERE fully dried hanging.

[server_log.txt](https://github.com/penguin0616/insight/files/5045710/server_log.txt)
video capture of the bug
https://youtu.be/RPa4-fDovh0

[00:00:56]: [string "../mods/workshop-2189004162/scripts/descrip..."]:12: attempt to perform arithmetic on field 'targettime' (a nil value)
LUA ERROR stack traceback:
../mods/workshop-2189004162/scripts/descriptors/dryer.lua:12 in (local) descriptor (Lua) <2-27>
   self =
      foodtype = MEAT
      ondonedrying = function - scripts/prefabs/meatrack.lua:99
      task = PERIODIC 103307: 8117.699923
      onharvest = function - scripts/prefabs/meatrack.lua:109
      ingredientperish = 0.41739326934289
      onstartdrying = function - scripts/prefabs/meatrack.lua:88
      inst = 103307 - meatrack (valid:true)
      _ = table: 22C31630
      tasktotime = 8117.6999226921
   context = table: 06331DA8
   inst = 103307 - meatrack (valid:true)
   description = nil
../mods/workshop-2189004162/modmain.lua:454 in (global) GetItemInformation (Lua) <418-518>
   item = 103307 - meatrack (valid:true)
   player = 123780 - wickerbottom (valid:true)
   player_context = table: 06331DA8
   chunks = table: 1DD73B08
   name = dryer
   component = table: 22C31A68
   descriptor = function - ../mods/workshop-2189004162/scripts/descriptors/dryer.lua:2
../mods/workshop-2189004162/modmain.lua:552 in (global) RequestItemInformation (Lua) <521-576>
   item = 103307 - meatrack (valid:true)
   player = 123780 - wickerbottom (valid:true)
   id = 102076
../mods/workshop-2189004162/modmain.lua:1001 in () ? (Lua) <999-1002>
   player = 123780 - wickerbottom (valid:true)
   entity = 103307 - meatrack (valid:true)
   id = 102076
=[C]:-1 in (method) CallRPC (C) <-1--1>
scripts/networkclientrpc.lua:946 in (global) HandleRPCQueue (Lua) <937-954>
   i = 1
   fn = function - ../mods/workshop-2189004162/modmain.lua:999
   sender = 123780 - wickerbottom (valid:true)
   data = table: 1DD736F8
   tick = true
scripts/update.lua:41 in () ? (Lua) <33-129>
   dt = 0.016660500317812

[00:00:56]: [string "../mods/workshop-2189004162/scripts/descrip..."]:12: attempt to perform arithmetic on field 'targettime' (a nil value)
LUA ERROR stack traceback:
    ../mods/workshop-2189004162/scripts/descriptors/dryer.lua:12 in (local) descriptor (Lua) <2-27>
    ../mods/workshop-2189004162/modmain.lua:454 in (global) GetItemInformation (Lua) <418-518>
    ../mods/workshop-2189004162/modmain.lua:552 in (global) RequestItemInformation (Lua) <521-576>
    ../mods/workshop-2189004162/modmain.lua:1001 in () ? (Lua) <999-1002>
    =[C]:-1 in (method) CallRPC (C) <-1--1>
    scripts/networkclientrpc.lua:946 in (global) HandleRPCQueue (Lua) <937-954>
    scripts/update.lua:41 in () ? (Lua) <33-129>
	
[00:00:56]: Warning: Widget:SetFocusFromChild is happening on a widget outside of the screen/widget hierachy. This will cause focus moves to fail. Is 	ScriptErrorWidget	not a screen?	
[00:00:56]: stack traceback:
	scripts/widgets/widget.lua:605 in (method) SetFocusFromChild (Lua) <602-627>
	scripts/widgets/widget.lua:624 in (method) SetFocusFromChild (Lua) <602-627>
	scripts/widgets/widget.lua:624 in (method) SetFocusFromChild (Lua) <602-627>
	scripts/widgets/widget.lua:656 in (method) SetFocus (Lua) <629-665>
	scripts/widgets/menu.lua:83 in (method) SetFocus (Lua) <74-85>
	scripts/widgets/scripterrorwidget.lua:109 in (method) OnUpdate (Lua) <102-119>
	scripts/update.lua:90 in () ? (Lua) <33-129>

**Comment by jkramr** [comment]
Commented out some code, played around with dryer.lua and got it working without crash. But containers are still not highlighting for me. You can use the video in the description above to see what mods were enabled in that build. Could be API Gem Core.

**Comment by jkramr** [comment]
```
modmain.lua:545
		if id == "yep" then
			id = item.GUID
		else
			error("[insight]: missing guid")
		end
```

this line of code is crashing everything instantly for me with [insight]: missing guid error in server log. hover up on literally any item crashes the game

**Comment by penguin0616** [comment]
That second one shouldn't have made it into the public version, my bad. Was debugging Wortox souls. I should have a fix for the drying rack soon enough.

**Comment by jkramr** [comment]
Thanks! Created a PR with a quick fix: #9 
Just in case, since I already fixed it for myself I figured I might as well share. Ignore if you fixed it already.

**Comment by penguin0616** [comment]
I have no idea how to actually do pull requests, but I think I pulled it off.
Somehow I didn't realize :GetTimeToDry, among the other functions, existed.
Thanks!
